{
  "data": {
    "repository": {
      "issue": {
        "number": 53,
        "title": "[Windows] : Couldn't find equation file! ",
        "body": "Hi Miles,\r\n\r\nI've been installing PySR in parallel to Julia under win10. It runs... till the moment it crashes with the following message:\r\n\r\n>  File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 774, in get_hof\r\n    raise RuntimeError(\"Couldn't find equation file! The equation search likely exited before a single iteration completed.\")\r\n\r\n> RuntimeError: Couldn't find equation file! The equation search likely exited before a single iteration completed.\r\n\r\nIn the last case, I've been to 38% of progress.\r\n\r\nI have to say that, sometime (not often), the process gets completed.\r\n\r\nWhat is the reason for this?\r\n\r\nAlso... is there a forum or I posted at the right place?\r\n\r\nI thank you for your help.\r\n\r\nRegards\r\n\r\nMagaud ",
        "comments": {
          "nodes": [
            {
              "author": {
                "login": "yxie20"
              },
              "body": "Similar issue here on Ubuntu. This happens only sometimes, and seems to be new in 0.6.3 (maybe).\r\n```\r\n==============================\r\n\r\nCycles per second: 6.670e+02\r\nHead worker occupation: 2.1%\r\nProgress: 1 / 4 total iterations (25.000%)\r\n==============================\r\nBest equations for output 1\r\nHall of Fame:\r\n-----------------------------------------\r\nComplexity  Loss       Score     Equation\r\n\r\n==============================\r\nBest equations for output 2\r\nHall of Fame:\r\n-----------------------------------------\r\nComplexity  Loss       Score     Equation\r\n2           3.983e+01  -5.284e-02  cos(0.51271415)\r\n3           1.090e+01  1.296e+00  (x1 * x1)\r\n5           9.370e+00  7.560e-02  ((x1 * x1) + -0.99658316)\r\n7           9.341e+00  1.560e-03  ((x1 * (x1 + 0.03651586)) + -0.99658316)\r\n8           8.388e+00  1.076e-01  (((x1 * x1) + -1.4183189) + cos(x4))\r\n10          8.339e+00  2.951e-03  (((x1 * x1) + -1.4183189) + cos(x4 + -0.24970347))\r\n12          8.180e+00  9.621e-03  ((-1.5824815 + (x1 * x1)) + (1.5988172 * cos(-0.24577418 + x4)))\r\n14          8.100e+00  4.877e-03  ((-1.725914 + (x1 * x1)) + (1.712476 * cos(0.19201946 + (-0.86918753 * x4))))\r\n\r\n==============================\r\nTraceback (most recent call last):\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pysr/sr.py\", line 782, in get_hof\r\n    all_outputs = [pd.read_csv(str(equation_file) + f'.out{i}' + '.bkup', sep=\"|\") for i in range(1, nout+1)]\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pysr/sr.py\", line 782, in <listcomp>\r\n    all_outputs = [pd.read_csv(str(equation_file) + f'.out{i}' + '.bkup', sep=\"|\") for i in range(1, nout+1)]\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pandas/io/parsers.py\", line 610, in read_csv\r\n    return _read(filepath_or_buffer, kwds)\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pandas/io/parsers.py\", line 462, in _read\r\n    parser = TextFileReader(filepath_or_buffer, **kwds)\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pandas/io/parsers.py\", line 819, in __init__\r\n    self._engine = self._make_engine(self.engine)\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pandas/io/parsers.py\", line 1050, in _make_engine\r\n    return mapping[engine](self.f, **self.options)  # type: ignore[call-arg]\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pandas/io/parsers.py\", line 1867, in __init__\r\n    self._open_handles(src, kwds)\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pandas/io/parsers.py\", line 1368, in _open_handles\r\n    storage_options=kwds.get(\"storage_options\", None),\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pandas/io/common.py\", line 647, in get_handle\r\n    newline=\"\",\r\nFileNotFoundError: [Errno 2] No such file or directory: 'hall_of_fame_2021-06-08_114346.574.csv.out1.bkup'\r\n\r\nDuring handling of the above exception, another exception occurred:\r\n\r\nTraceback (most recent call last):\r\n  File \"optimal.py\", line 21, in <module>\r\n    julia_optimization=0,       # Faster startup time. Turn off optimizing compiler for Julia code\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pysr/sr.py\", line 395, in pysr\r\n    equations = get_hof(**kwargs)\r\n  File \"/home/yxie20/anaconda3/envs/rw/lib/python3.7/site-packages/pysr/sr.py\", line 786, in get_hof\r\n    raise RuntimeError(\"Couldn't find equation file! The equation search likely exited before a single iteration completed.\")\r\nRuntimeError: Couldn't find equation file! The equation search likely exited before a single iteration completed.\r\n```",
              "createdAt": "2021-06-08T15:50:03Z"
            },
            {
              "author": {
                "login": "yxie20"
              },
              "body": "Another possibly related error:\r\n```\r\n==============================\r\n\r\nCycles per second: 2.290e+03\r\nHead worker occupation: 2.5%\r\nProgress: 2 / 4 total iterations (50.000%)\r\n==============================\r\nBest equations for output 1\r\nHall of Fame:\r\n-----------------------------------------\r\nComplexity  Loss       Score     Equation\r\n1           4.739e+01  -1.096e-02  1.8519074\r\n3           2.195e+01  3.848e-01  (x0 * x0)\r\n5           2.017e+01  4.236e-02  ((x0 * x0) + -1.4364125)\r\n9           2.090e+00  5.667e-01  (((x0 * x0) + (x1 * x2)) + -1.7970624)\r\n11          2.061e+00  6.839e-03  (((x0 * (x0 + 0.079849906)) + (x1 * x2)) + -1.7017426)\r\n12          5.211e-01  1.375e+00  (((-1.8610494 + cos(x3)) + (x0 * x0)) + (x1 * x2))\r\n14          7.465e-11  1.133e+01  (((-2.0000072 + (x0 * x0)) + (x1 * x2)) + (2.0000086 * cos(x3)))\r\n\r\n==============================\r\nBest equations for output 2\r\nHall of Fame:\r\n-----------------------------------------\r\nComplexity  Loss       Score     Equation\r\n1           9.285e+01  -6.034e-02  1.5331783\r\n2           8.836e+01  4.952e-02  exp(1.0611964)\r\n3           2.174e+01  1.402e+00  (x1 * x1)\r\n5           1.989e+01  4.451e-02  ((x1 * x1) + -1.3825463)\r\n7           1.970e+01  4.892e-03  (((x1 + -0.18729796) * x1) + -1.3825463)\r\n9           1.963e+01  1.840e-03  ((((x1 + -0.19800864) * x1) + -1.5256172) * 1.033466)\r\n\r\n==============================\r\n┌ Error: Fatal error on process 4\r\n│   exception =\r\n│    IOError: stream is closed or unusable\r\n│    Stacktrace:\r\n│      [1] check_open\r\n│        @ ./stream.jl:386 [inlined]\r\n│      [2] uv_write_async(s::Sockets.TCPSocket, p::Ptr{UInt8}, n::UInt64)\r\n│        @ Base ./stream.jl:1018\r\n│      [3] uv_write(s::Sockets.TCPSocket, p::Ptr{UInt8}, n::UInt64)\r\n│        @ Base ./stream.jl:981\r\n│      [4] uv_write\r\n│        @ ./stream.jl:977 [inlined]\r\n│      [5] unsafe_write(s::Sockets.TCPSocket, p::Ptr{UInt8}, n::UInt64)\r\n│        @ Base ./stream.jl:1061\r\n│      [6] unsafe_write\r\n│        @ ./io.jl:646 [inlined]\r\n│      [7] unsafe_write(s::Sockets.TCPSocket, p::Base.RefValue{Float32}, n::Int64)\r\n│        @ Base ./io.jl:644\r\n│      [8] write\r\n│        @ ./io.jl:647 [inlined]\r\n│      [9] write\r\n│        @ ./io.jl:650 [inlined]\r\n│     [10] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, n::Float32)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:613\r\n│     [11] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [12] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [13] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [14] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [15] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [16] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [17] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [18] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [19] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [20] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [21] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [22] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [23] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [24] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [25] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [26] serialize\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636 [inlined]\r\n│     [27] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, a::Vector{PopMember{Float32}})\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:273\r\n│     [28] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [29] serialize\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636 [inlined]\r\n│     [30] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, t::Tuple{Population{Float32}, HallOfFame, Dict{String, Any}})\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:201\r\n│     [31] serialize_msg(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, o::Distributed.ResultMsg)\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:78\r\n│     [32] #invokelatest#2\r\n│        @ ./essentials.jl:708 [inlined]\r\n│     [33] invokelatest\r\n│        @ ./essentials.jl:706 [inlined]\r\n│     [34] send_msg_(w::Distributed.Worker, header::Distributed.MsgHeader, msg::Distributed.ResultMsg, now::Bool)\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:174\r\n│     [35] send_msg_now\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:118 [inlined]\r\n│     [36] send_msg_now(s::Sockets.TCPSocket, header::Distributed.MsgHeader, msg::Distributed.ResultMsg)\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:113\r\n│     [37] deliver_result(sock::Sockets.TCPSocket, msg::Symbol, oid::Distributed.RRID, value::Tuple{Population{Float32}, HallOfFame, Dict{String, Any}})\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/process_messages.jl:95\r\n│     [38] macro expansion\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/process_messages.jl:286 [inlined]\r\n│     [39] (::Distributed.var\"#105#107\"{Distributed.CallMsg{:call_fetch}, Distributed.MsgHeader, Sockets.TCPSocket})()\r\n│        @ Distributed ./task.jl:411\r\n└ @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/process_messages.jl:99\r\n┌ Error: Fatal error on process 3\r\n│   exception =\r\n│    IOError: stream is closed or unusable\r\n│    Stacktrace:\r\n│      [1] check_open\r\n│        @ ./stream.jl:386 [inlined]\r\n│      [2] uv_write_async(s::Sockets.TCPSocket, p::Ptr{UInt8}, n::UInt64)\r\n│        @ Base ./stream.jl:1018\r\n│      [3] uv_write(s::Sockets.TCPSocket, p::Ptr{UInt8}, n::UInt64)\r\n│        @ Base ./stream.jl:981\r\n│      [4] uv_write\r\n│        @ ./stream.jl:977 [inlined]\r\n│      [5] unsafe_write(s::Sockets.TCPSocket, p::Ptr{UInt8}, n::UInt64)\r\n│        @ Base ./stream.jl:1061\r\n│      [6] unsafe_write\r\n│        @ ./io.jl:646 [inlined]\r\n│      [7] unsafe_write(s::Sockets.TCPSocket, p::Base.RefValue{UInt16}, n::Int64)\r\n│        @ Base ./io.jl:644\r\n│      [8] write\r\n│        @ ./io.jl:647 [inlined]\r\n│      [9] write\r\n│        @ ./io.jl:650 [inlined]\r\n│     [10] serialize_cycle(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:156\r\n│     [11] serialize_type_data(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, t::DataType)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:542\r\n│     [12] serialize_type(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, t::DataType, ref::Bool)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:583\r\n│     [13] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:651\r\n│     [14] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [15] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [16] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [17] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [18] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [19] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [20] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [21] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [22] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [23] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [24] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [25] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [26] serialize\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636 [inlined]\r\n│     [27] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, a::Vector{PopMember{Float32}})\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:273\r\n│     [28] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [29] serialize\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636 [inlined]\r\n│     [30] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, t::Tuple{Population{Float32}, HallOfFame, Dict{String, Any}})\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:201\r\n│     [31] serialize_msg(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, o::Distributed.ResultMsg)\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:78\r\n│     [32] #invokelatest#2\r\n│        @ ./essentials.jl:708 [inlined]\r\n│     [33] invokelatest\r\n│        @ ./essentials.jl:706 [inlined]\r\n│     [34] send_msg_(w::Distributed.Worker, header::Distributed.MsgHeader, msg::Distributed.ResultMsg, now::Bool)\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:174\r\n│     [35] send_msg_now\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:118 [inlined]\r\n│     [36] send_msg_now(s::Sockets.TCPSocket, header::Distributed.MsgHeader, msg::Distributed.ResultMsg)\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:113\r\n│     [37] deliver_result(sock::Sockets.TCPSocket, msg::Symbol, oid::Distributed.RRID, value::Tuple{Population{Float32}, HallOfFame, Dict{String, Any}})\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/process_messages.jl:95\r\n│     [38] macro expansion\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/process_messages.jl:286 [inlined]\r\n│     [39] (::Distributed.var\"#105#107\"{Distributed.CallMsg{:call_fetch}, Distributed.MsgHeader, Sockets.TCPSocket})()\r\n│        @ Distributed ./task.jl:411\r\n└ @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/process_messages.jl:99\r\n```",
              "createdAt": "2021-06-08T16:30:42Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "The message `Couldn't find equation file! The equation search likely exited before a single iteration completed.` means that some error occurred earlier, and PySR couldn't recover the equation file from the run. Did it print any message before that?\r\n\r\nAlso, does this occur on 0.6.4 or just 0.6.3? 0.6.4 was a same-day hotfix I pushed for errors related to to NaNs/Infs not being checked in the equation evaluation, which might be related.\r\n\r\n@yxie20 does that error crash your run, or is it just a message that gets printed after? Sometimes if the Julia processes are not cleanly shut down, they will print these messages, but they are not actually errors that affect the pipeline. Btw I would strongly recommend using a larger number of iterations if possible, 4 is quite small.",
              "createdAt": "2021-06-08T16:41:21Z"
            },
            {
              "author": {
                "login": "Magaud59"
              },
              "body": "Hi Miles,\r\n\r\nso, first thing... I update the package from 0.6.0 to 0.6.4. \r\nFollowing the update, I ran my code again to get the full error message. Here it is:\r\n\r\n> Traceback (most recent call last):\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 784, in get_hof\r\n>     all_outputs = [pd.read_csv(str(equation_file) + '.bkup', sep=\"|\")]\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 686, in read_csv\r\n>     return _read(filepath_or_buffer, kwds)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 452, in _read\r\n>     parser = TextFileReader(fp_or_buf, **kwds)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 946, in __init__\r\n>     self._make_engine(self.engine)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 1178, in _make_engine\r\n>     self._engine = CParserWrapper(self.f, **self.options)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 2008, in __init__\r\n>     self._reader = parsers.TextReader(src, **kwds)\r\n> \r\n>   File \"pandas\\_libs\\parsers.pyx\", line 382, in pandas._libs.parsers.TextReader.__cinit__\r\n> \r\n>   File \"pandas\\_libs\\parsers.pyx\", line 674, in pandas._libs.parsers.TextReader._setup_parser_source\r\n> \r\n> FileNotFoundError: [Errno 2] No such file or directory: 'hall_of_fame_2021-06-08_200303.084.csv.bkup'\r\n> \r\n> \r\n> During handling of the above exception, another exception occurred:\r\n> \r\n> Traceback (most recent call last):\r\n> \r\n>   File \"C:\\Users\\Matthieu\\Documents\\Project_Work\\Non-Contact\\VirusScan\\RF_Models\\LineCalcRegression_.py\", line 19, in <module>\r\n>     equations = pysr(X, y,niterations=20,\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 395, in pysr\r\n>     equations = get_hof(**kwargs)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 786, in get_hof\r\n>     raise RuntimeError(\"Couldn't find equation file! The equation search likely exited before a single iteration completed.\")\r\n> \r\n> RuntimeError: Couldn't find equation file! The equation search likely exited before a single iteration completed.\r\n\r\nLet me know if I could support you in another way.\r\n\r\nRegards\r\n\r\nMagaud",
              "createdAt": "2021-06-09T07:06:06Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "Thanks. What is the error before this? These errors about not finding the equation file are just what happens what PySR crashed, but don't specify how it crashed. There should be a message earlier about the original error.\r\n\r\nAlso, can you show what your PySR script is?",
              "createdAt": "2021-06-09T07:51:40Z"
            },
            {
              "author": {
                "login": "Magaud59"
              },
              "body": "That's the only error that is reported. The line before that are just the standard feedback report:\r\n\r\n> ==============================\r\n> \r\n> Cycles per second: 6.000e+03\r\n> Head worker occupation: 2.8%\r\n> Progress: 130 / 400 total iterations (32.500%)\r\n> ==============================\r\n> Hall of Fame:\r\n> -----------------------------------------\r\n> Complexity  Loss       Score     Equation\r\n> 1           8.823e+02  3.982e-01  x0\r\n> 3           1.315e+00  3.254e+00  pow(x0, 0.8966019)\r\n> 5           1.724e-01  1.016e+00  (0.78418094 * pow(x0, 0.9433791))\r\n> 7           1.721e-01  7.639e-04  (0.014364422 + (0.7841787 * pow(x0, 0.9433453)))\r\n> 11          1.223e-01  8.550e-02  (erf(erf(-0.073428765 * x0)) + pow(0.86148334 * x0, 0.9246638))\r\n> \r\n> ==============================\r\n\r\nNothing else...",
              "createdAt": "2021-06-09T08:30:11Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "That’s very strange. It should definitely be able to find the equation file then, I think. Can I see your PySR call?",
              "createdAt": "2021-06-09T13:35:37Z"
            },
            {
              "author": {
                "login": "Magaud59"
              },
              "body": "I'm not sure to know what it is... sorry for that...",
              "createdAt": "2021-06-09T13:54:15Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "What I mean is: can I see the code you are running? This one: `C:\\Users\\Matthieu\\Documents\\Project_Work\\Non-Contact\\VirusScan\\RF_Models\\LineCalcRegression_.py`. Fine if you delete the code that is private - I am just curious to see the line with `pysr(...)`.",
              "createdAt": "2021-06-09T15:12:41Z"
            },
            {
              "author": {
                "login": "Magaud59"
              },
              "body": "No Problem...\r\n\r\nhere it is:\r\n\r\n> equations = pysr(X, y,niterations=20,\r\n>             binary_operators=[\"plus\", \"mult\",\"pow\",\"div\"],\r\n>             unary_operators=[ \"exp\", \"sqrt_abs\", \"log_abs\",\"erf\",\"erfc\"])\r\n> \r\n> print(equations)\r\n\r\n",
              "createdAt": "2021-06-09T15:17:07Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "For debugging purposes, can you set procs=0, and also remove \"erf\" and \"erfc\"? I'm curious to see what the output is then. Thanks!",
              "createdAt": "2021-06-09T17:08:51Z"
            },
            {
              "author": {
                "login": "amariad"
              },
              "body": "Hi. I'm on mac OS 10 and I'm getting the same error. I've tried different versions of pysr and python but continue to get some sort of version conflicts, even when only testing on the short example code. \r\n\r\nHere's an example of the errors in my terminal:\r\n\r\nI will install Julia packages using PySR's Project.toml file. OK? (y/n): y\r\nOK. I will install at launch.\r\nRunning on julia -O3 /var/folders/24/lpb6cn7d55xf7plwphkv0pfm0000gn/T/tmpgse65yqn/runfile.jl\r\n  Activating environment at `/Applications/anaconda3/envs/test/lib/python3.7/site-packages/Project.toml`\r\nERROR: LoadError: Unsatisfiable requirements detected for package AbstractAlgebra [c3fe647b]:\r\n AbstractAlgebra [c3fe647b] log:\r\n ├─AbstractAlgebra [c3fe647b] has no known versions!\r\n └─found to have no compatible versions left with SymbolicUtils [d1185830]\r\n   └─SymbolicUtils [d1185830] log:\r\n     ├─possible versions are: 0.1.0-0.11.3 or uninstalled\r\n     └─restricted by compatibility requirements with SymbolicRegression [8254be44] to versions: 0.6.0-0.6.3\r\n       └─SymbolicRegression [8254be44] log:\r\n         ├─possible versions are: 0.1.0-0.6.1 or uninstalled\r\n         └─restricted to versions 0.6.1-0.6 by an explicit requirement, leaving only versions 0.6.1\r\n\r\nMany thanks for the help.",
              "createdAt": "2021-06-09T18:25:42Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "@amariad this looks like your julia packages might be out of date. Can you try the solution here? https://github.com/MilesCranmer/PySR/issues/27\r\n\r\nMake sure to run `pip install --upgrade pysr` too - it looks like you might be on 0.6.1",
              "createdAt": "2021-06-09T19:36:24Z"
            },
            {
              "author": {
                "login": "amariad"
              },
              "body": "@MilesCranmer, thanks! Solution #27 worked. I was using version 0.6.1 because version 0.6.6 had been giving me and another collaborator some problems. But all seems to be working now!",
              "createdAt": "2021-06-09T21:17:27Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "Awesome!!! I should add that trick to the README, my sense is that issues related to it come up a lot.",
              "createdAt": "2021-06-09T21:25:39Z"
            },
            {
              "author": {
                "login": "Magaud59"
              },
              "body": "Hi Miles,\r\n\r\nsorry, It took a bit of time but here are the feedback of the tests:\r\n\r\nwith procs=0:\r\n\r\n> ==============================\r\n> \r\n> Cycles per second: 1.750e+03\r\n> Head worker occupation: 99.8%\r\n> Progress: 245 / 400 total iterations (61.250%)\r\n> ==============================\r\n> Hall of Fame:\r\n> -----------------------------------------\r\n> Complexity  Loss       Score     Equation\r\n> 1           8.823e+02  3.982e-01  x0\r\n> 3           1.315e+00  3.254e+00  pow(x0, 0.8966013)\r\n> 5           1.724e-01  1.016e+00  pow(0.7728203 * x0, 0.94337857)\r\n> 15          1.677e-01  2.749e-03  pow(0.77275 * (x0 + sqrt_abs(-0.0061262175 * (1.0 / sqrt_abs(2.0 * x0)))), 0.94336826)\r\n> \r\n> ==============================\r\n> Traceback (most recent call last):\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 784, in get_hof\r\n>     all_outputs = [pd.read_csv(str(equation_file) + '.bkup', sep=\"|\")]\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 686, in read_csv\r\n>     return _read(filepath_or_buffer, kwds)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 452, in _read\r\n>     parser = TextFileReader(fp_or_buf, **kwds)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 946, in __init__\r\n>     self._make_engine(self.engine)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 1178, in _make_engine\r\n>     self._engine = CParserWrapper(self.f, **self.options)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 2008, in __init__\r\n>     self._reader = parsers.TextReader(src, **kwds)\r\n> \r\n>   File \"pandas\\_libs\\parsers.pyx\", line 382, in pandas._libs.parsers.TextReader.__cinit__\r\n> \r\n>   File \"pandas\\_libs\\parsers.pyx\", line 674, in pandas._libs.parsers.TextReader._setup_parser_source\r\n> \r\n> FileNotFoundError: [Errno 2] No such file or directory: 'hall_of_fame_2021-06-09_191317.555.csv.bkup'\r\n> \r\n> \r\n> During handling of the above exception, another exception occurred:\r\n> \r\n> Traceback (most recent call last):\r\n> \r\n>   File \"C:\\Users\\Matthieu\\Documents\\Project_Work\\Non-Contact\\VirusScan\\RF_Models\\LineCalcRegression_.py\", line 19, in <module>\r\n>     equations = pysr(X, y,procs=0, niterations=20,\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 395, in pysr\r\n>     equations = get_hof(**kwargs)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 786, in get_hof\r\n>     raise RuntimeError(\"Couldn't find equation file! The equation search likely exited before a single iteration completed.\")\r\n> \r\n> RuntimeError: Couldn't find equation file! The equation search likely exited before a single iteration completed.\r\n> \r\n> \r\n> -------------------------------------------------------------------------------------\r\n> \r\n\r\n\r\nwithout ERF and EFRC functions:\r\n\r\n\r\n> \r\n> ==============================\r\n> \r\n> Cycles per second: 5.960e+03\r\n> Head worker occupation: 2.4%\r\n> Progress: 47 / 400 total iterations (11.750%)\r\n> ==============================\r\n> Hall of Fame:\r\n> -----------------------------------------\r\n> Complexity  Loss       Score     Equation\r\n> 1           8.823e+02  3.982e-01  x0\r\n> 3           1.315e+00  3.254e+00  pow(x0, 0.89659876)\r\n> 5           1.724e-01  1.016e+00  pow(-0.7731549 * x0, 0.9432982)\r\n> \r\n> ==============================\r\n> Traceback (most recent call last):\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 784, in get_hof\r\n>     all_outputs = [pd.read_csv(str(equation_file) + '.bkup', sep=\"|\")]\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 686, in read_csv\r\n>     return _read(filepath_or_buffer, kwds)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 452, in _read\r\n>     parser = TextFileReader(fp_or_buf, **kwds)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 946, in __init__\r\n>     self._make_engine(self.engine)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 1178, in _make_engine\r\n>     self._engine = CParserWrapper(self.f, **self.options)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\", line 2008, in __init__\r\n>     self._reader = parsers.TextReader(src, **kwds)\r\n> \r\n>   File \"pandas\\_libs\\parsers.pyx\", line 382, in pandas._libs.parsers.TextReader.__cinit__\r\n> \r\n>   File \"pandas\\_libs\\parsers.pyx\", line 674, in pandas._libs.parsers.TextReader._setup_parser_source\r\n> \r\n> FileNotFoundError: [Errno 2] No such file or directory: 'hall_of_fame_2021-06-10_080002.191.csv.bkup'\r\n> \r\n> \r\n> During handling of the above exception, another exception occurred:\r\n> \r\n> Traceback (most recent call last):\r\n> \r\n>   File \"C:\\Users\\Matthieu\\Documents\\Project_Work\\Non-Contact\\VirusScan\\RF_Models\\LineCalcRegression_.py\", line 19, in <module>\r\n>     equations = pysr(X, y, niterations=20,\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 395, in pysr\r\n>     equations = get_hof(**kwargs)\r\n> \r\n>   File \"C:\\Users\\Matthieu\\anaconda3\\lib\\site-packages\\pysr\\sr.py\", line 786, in get_hof\r\n>     raise RuntimeError(\"Couldn't find equation file! The equation search likely exited before a single iteration completed.\")\r\n> \r\n> RuntimeError: Couldn't find equation file! The equation search likely exited before a single iteration completed.\r\n\r\nI hope that this helps.\r\n\r\nAll the best.\r\n\r\nMatthieu",
              "createdAt": "2021-06-10T07:18:38Z"
            },
            {
              "author": {
                "login": "yxie20"
              },
              "body": "> The message `Couldn't find equation file! The equation search likely exited before a single iteration completed.` means that some error occurred earlier, and PySR couldn't recover the equation file from the run. Did it print any message before that?\r\n> \r\n> Also, does this occur on 0.6.4 or just 0.6.3? 0.6.4 was a same-day hotfix I pushed for errors related to to NaNs/Infs not being checked in the equation evaluation, which might be related.\r\n> \r\n> @yxie20 does that error crash your run, or is it just a message that gets printed after? Sometimes if the Julia processes are not cleanly shut down, they will print these messages, but they are not actually errors that affect the pipeline. Btw I would strongly recommend using a larger number of iterations if possible, 4 is quite small.\r\n\r\nCrashes the run. \r\n\r\nWhat's the drawback with a small number of `iterations`? I've found `npop` and `ncyclesperiteration` to be the two most important hyperparameters for finding the ground truth equation.\r\n\r\nThank you.",
              "createdAt": "2021-06-10T13:42:12Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "@Magaud59 this is strange. If it doesn't print other error messages, I am really not sure what it could be. Can you run the example in the README without issue?\r\n\r\nIf the example in the README runs fine, then I think it is a bug internally. Can you try adding `sub` as an operator? I wonder if there is some bug because you have `plus` but no subtraction - maybe the simplification backend, even though it specifically has checks against this, tries to create a subtraction operator and this errors.\r\n\r\nBy the way - `pow` and `sqrt_abs` , and also `pow` and `exp` are redundant operations, so I would only choose one out of each to make the search more efficient.\r\n\r\nMore generally, and for @yxie20, if you can give me an [MWE](https://en.wikipedia.org/wiki/Minimal_working_example#:~:text=In%20computing%2C%20a%20minimal%20working,to%20be%20demonstrated%20and%20reproduced.&text=A%20minimal%20working%20example%20may,short%20self%2Dcontained%20correct%20example.) that I can run on my own machine, that would make this process faster since I can quickly debug these different choices. \r\n\r\n@yxie20 re: those kwargs, they are all sort of related. `niterations * npop * ncyclesperiteration` is the total number of generations in the evolutionary algorithm. `ncyclesperiteration` sets the number of generations between each migration period. `npop` is the number of equations within each population.\r\n",
              "createdAt": "2021-06-10T17:54:47Z"
            },
            {
              "author": {
                "login": "yxie20"
              },
              "body": "Found the problem: the bug happens when `niterations=1` \r\n\r\nI'm using some ridiculous hyperparameters. Although I still don't have a good grasp of the meaning behind hyperparameters after reading the documentation and the original paper... Your comments have been helpful, however!\r\n\r\nMWE:\r\n```\r\nimport numpy as np\r\nfrom pysr import pysr, best\r\n\r\n# Dataset (alternative)\r\nX = 2*np.random.randn(100, 8)\r\ny = 2*np.cos(X[:, 3:6]) + X[:, 0:3]**2 - 2 + X[:,2:5]*X[:,1:4]\r\n\r\n\r\nequations = pysr(X, y, \r\n    binary_operators=[\"plus\", \"mult\"],\r\n    unary_operators=[\"cos\", \"exp\", \"sin\"], \r\n    # procs=4,\r\n    maxsize=5,\r\n    populations=1,             # Number of populations running.\r\n    npop=100,                  # Number of individuals per population. More population, slower, more chance of hitting the correct.\r\n    niterations=3,              # Iterations per population.  populations * niterations = progress bar\r\n    ncyclesperiteration=50,    # Number of total mutations per 10 samples of population each iteration\r\n    annealing=True,             # With False, simple equations take longer but more complex equations are achievable\r\n    julia_optimization=0,       # Faster startup time. Turn off optimizing compiler for Julia code\r\n    temp_equation_file=True,\r\n    # fast_cycle=True,          # True didn't let the algo find the true equation.\r\n)\r\n```",
              "createdAt": "2021-06-10T18:21:47Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "Hi @yxie20, thanks, this is quite helpful. I should first mention that indeed PySR needs a full paper out to help describe all of these options. This paper should be up later this month. This page will also be helpful if you haven't seen it: https://pysr.readthedocs.io/en/latest/docs/options/.\r\n\r\nFirst - a maxsize of 5 is very very small. I think the initial population is randomly sampled from maxsize=5. There is basically no room for the optimization algorithm to move around. Your ground truth equation has a complexity of 13, so it will be impossible to recover it (and generally it helps to have a maxsize larger than your ground truth, so there is \"room\" to move around).\r\n\r\nSecond - using `plus` but not `sub` will not allow PySR to do any equation simplification, which tends to help the search a lot. \r\n\r\nHowever, I can still run your script without crashing. Are you on PySR >= 0.6.4?",
              "createdAt": "2021-06-10T18:36:22Z"
            },
            {
              "author": {
                "login": "yxie20"
              },
              "body": "Thank you, Miles! I'll definitely look forward to your full paper (and congratulations in advance!)\r\n\r\nThank you for that link! Indeed I missed it. Reading now. `maxsize=5` is definitely ridiculous... agreed! I set it small for fast debugging.\r\n\r\nLooks like the bug went away on 0.6.6. The Julia install bug #27 also went away today (strange), but I won't complain!",
              "createdAt": "2021-06-10T18:44:09Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "Great that it works! By the way - I can recover your ground truth equations exactly via the following setup:\r\n\r\n```python\r\nimport numpy as np\r\nfrom pysr import pysr, best\r\n\r\n# Dataset (alternative)\r\nX = 2*np.random.randn(100, 8)\r\ny = 2*np.cos(X[:, 3:6]) + X[:, 0:3]**2 - 2 + X[:,2:5]*X[:,1:4]\r\n\r\nequations = pysr(X, y,\r\n    binary_operators='+ * / -'.split(' '),\r\n    unary_operators=[\"cos\", \"exp\", \"sin\"],\r\n    niterations=10,\r\n    procs=4,\r\n    maxsize=20,\r\n    julia_optimization=3,\r\n    temp_equation_file=True,\r\n)\r\n\r\nprint(equations)\r\n```\r\n\r\nHere is the output:\r\n\r\n<img width=\"654\" alt=\"Screen Shot 2021-06-10 at 2 52 11 PM\" src=\"https://user-images.githubusercontent.com/7593028/121581061-70b7cc80-c9fb-11eb-900b-4aeaf8db72cf.png\">\r\n\r\nRegarding this:\r\n> The Julia install bug #27 also went away today (strange)\r\n\r\nthis is actually expected. The Julia registry server takes a while to update. It gives you this bug when the new PySR version requires some SymbolicRegression.jl version that hasn't been included in the Julia registry yet. One fix is to wait for it to update as happened with your system, the other fix is to switch from the Julia registry server to the git-based registry (as on #27) which is instantly updated.\r\n",
              "createdAt": "2021-06-10T18:54:58Z"
            },
            {
              "author": {
                "login": "yxie20"
              },
              "body": "Unfortunately, I think there are more than one bug going on here... since \"Couldn't find equation file\" is an umbrella error message for any early-exits in Julia. These errors are always on-and-off, so I'm wrapping pysr code within try/except blocks in a while loop in my code--a quick fix.\r\n\r\nHere is one type of error (Julia's IOError):\r\n```\r\n  Activating environment at `~/anaconda3/envs/rw/lib/python3.7/site-packages/Project.toml`\r\n    Updating registry at `~/.julia/registries/General`\r\n  No Changes to `~/anaconda3/envs/rw/lib/python3.7/site-packages/Project.toml`\r\n  No Changes to `~/anaconda3/envs/rw/lib/python3.7/site-packages/Manifest.toml`\r\n  Activating environment at `~/anaconda3/envs/rw/lib/python3.7/site-packages/Project.toml`\r\n  Activating environment at `~/anaconda3/envs/rw/lib/python3.7/site-packages/Project.toml`\r\n  Activating environment at `~/anaconda3/envs/rw/lib/python3.7/site-packages/Project.toml`  Activating\r\n environment at `~/anaconda3/envs/rw/lib/python3.7/site-packages/Project.toml`\r\n┌ Error: Fatal error on process 2\r\n│   exception =\r\n│    IOError: stream is closed or unusable\r\n│    Stacktrace:\r\n│      [1] check_open\r\n│        @ ./stream.jl:386 [inlined]\r\n│      [2] uv_write_async(s::Sockets.TCPSocket, p::Ptr{UInt8}, n::UInt64)\r\n│        @ Base ./stream.jl:1018\r\n│      [3] uv_write(s::Sockets.TCPSocket, p::Ptr{UInt8}, n::UInt64)\r\n│        @ Base ./stream.jl:981\r\n│      [4] uv_write\r\n│        @ ./stream.jl:977 [inlined]\r\n│      [5] unsafe_write(s::Sockets.TCPSocket, p::Ptr{UInt8}, n::UInt64)\r\n│        @ Base ./stream.jl:1061\r\n│      [6] unsafe_write\r\n│        @ ./io.jl:646 [inlined]\r\n│      [7] unsafe_write(s::Sockets.TCPSocket, p::Base.RefValue{UInt8}, n::Int64)\r\n│        @ Base ./io.jl:644\r\n│      [8] write\r\n│        @ ./io.jl:647 [inlined]\r\n│      [9] write(s::Sockets.TCPSocket, b::UInt8)\r\n│        @ Base ./stream.jl:1102\r\n│     [10] writetag\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:142 [inlined]\r\n│     [11] serialize_cycle(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:155\r\n│     [12] serialize_type_data(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, t::DataType)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:542\r\n│     [13] serialize_type(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, t::DataType, ref::Bool)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:583\r\n│     [14] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:651\r\n│     [15] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [16] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [17] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [18] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [19] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636\r\n│     [20] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [21] serialize\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636 [inlined]\r\n│     [22] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, a::Vector{PopMember{Float32}})\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:273\r\n│     [23] serialize_any(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, x::Any)\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:657\r\n│     [24] serialize\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:636 [inlined]\r\n│     [25] serialize(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, t::Tuple{Population{Float32}, HallOfFame, Dict{String, Any}})\r\n│        @ Serialization /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Serialization/src/Serialization.jl:201\r\n│     [26] serialize_msg(s::Distributed.ClusterSerializer{Sockets.TCPSocket}, o::Distributed.ResultMsg)\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:78\r\n│     [27] #invokelatest#2\r\n│        @ ./essentials.jl:708 [inlined]\r\n│     [28] invokelatest\r\n│        @ ./essentials.jl:706 [inlined]\r\n│     [29] send_msg_(w::Distributed.Worker, header::Distributed.MsgHeader, msg::Distributed.ResultMsg, now::Bool)\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:174\r\n│     [30] send_msg_now\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:118 [inlined]\r\n│     [31] send_msg_now(s::Sockets.TCPSocket, header::Distributed.MsgHeader, msg::Distributed.ResultMsg)\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/messages.jl:113\r\n│     [32] deliver_result(sock::Sockets.TCPSocket, msg::Symbol, oid::Distributed.RRID, value::Tuple{Population{Float32}, HallOfFame, Dict{String, Any}})\r\n│        @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/process_messages.jl:95\r\n│     [33] macro expansion\r\n│        @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/process_messages.jl:286 [inlined]\r\n│     [34] (::Distributed.var\"#105#107\"{Distributed.CallMsg{:call_fetch}, Distributed.MsgHeader, Sockets.TCPSocket})()\r\n│        @ Distributed ./task.jl:411\r\n└ @ Distributed /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Distributed/src/process_messages.jl:99\r\n```\r\nMWE (although I don't think the parameter/data actually matters much here):\r\n```\r\nimport numpy as np\r\nfrom pysr import pysr, best\r\n\r\n# Dataset (alternative)\r\nX = 2*np.random.randn(512, 32)\r\ny = 2*np.cos(X[:, 3:6]) + X[:, 0:3]**2 - 2 + X[:,2:5]*X[:,1:4]\r\n\r\nequations = pysr(X, y, \r\n    binary_operators=[\"plus\", \"sub\", \"mult\", \"div\"],\r\n    unary_operators=[\"exp\", \"log_abs\", \"cos\", \"sin\"],\r\n    verbosity=0,\r\n    # procs=6,\r\n    temp_equation_file=True,\r\n    progress=False,\r\n    julia_optimization=0,       # Faster startup time. Turn off optimizing compiler for Julia code\r\n    output_torch_format=True,\r\n    annealing=True,            # With False, simple equations take longer but more complex equations are achievable\r\n    batching=True,\r\n    # # Optimal configs\r\n    niterations=2,              # Iterations per population of the entire algorithm. Best equations are printed and migrated between populations.  populations * niterations = progress bar. This doesnt really matter\r\n    maxsize=20,\r\n    populations=2,              # Number of populations running. (must > 1)\r\n    npop=2000,                  # Number of individuals per population. More population, slower, more chance of hitting the correct.\r\n)\r\n```",
              "createdAt": "2021-06-14T15:34:46Z"
            },
            {
              "author": {
                "login": "Magaud59"
              },
              "body": "Hi,\r\n\r\nI'm very sorry but I will let go on this point.\r\nIn any case, I thank you for the time you spent on my case.\r\n\r\nRegards\r\n\r\nM",
              "createdAt": "2021-06-14T15:42:49Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "@yxie20 ~~I am a bit confused because your MWE you have `procs=0`, but the error you shared is from multi-processing (will only appear for `procs>0`).~~ (edit: oops, sorry, you have `procs=4`) Are you sure that error is from that code, or is it delayed printing from a different process? The IOError is just from a process exiting if Julia closes due to an error. e.g., you see the same error if you just hit control-C when PySR is running in multi-processing mode. It can be very delayed printing if your single iteration runtime is long, and Julia closes without cleanly shutting down the process. You can run `pkill julia` at the command line to close hanging processes if necessary.",
              "createdAt": "2021-06-15T20:29:57Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "I’ve ran this code a few times, and get no errors. How often does it show up, and are you sure it’s an error from that run, or just a printout from processes closing late from the previous run?",
              "createdAt": "2021-06-16T10:50:20Z"
            },
            {
              "author": {
                "login": "MilesCranmer"
              },
              "body": "The original issue is unrelated, so closing this issue. But @yxie20 let me know how it goes.",
              "createdAt": "2021-06-16T21:48:56Z"
            }
          ],
          "pageInfo": {
            "hasNextPage": false,
            "endCursor": "Y3Vyc29yOnYyOpHOM2yG2w=="
          }
        }
      }
    }
  }
}